### PR TITLE
chore: Various template fixes

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
### Non-template Changes:
  - **Split the reason text into _author_ and _reason_**.
    _This reduces notification noise for the author, as before they were receiving notifications for both the PR and the commit which contained the reason text._
  - **Assign the generated PRs to the _author_ instead of `stackabletech/Developers`**.
    _This reduces notification noise for the team when it is likely that the author will be merging the changes._
  - **pre-commit/actionlint: Move the actionlint config to the correct location, and check templated files too**.

### Template changes (to be rolled out):
  - **pre-commit/actionlint: Move the actionlint config to the correct location**.
  - **pre-commit/workflow: Checkout git submodules**.
    _Needed for listener and secret operators._
  - **pre-commit/workflow: Install nix**.
    _Needed for running cargo commands via `nix-shell`_ 
  - **pre-commit: Run cargo commands via `nix-shell`**.
    _This ensures the necessary dependencies are there_. 
  - **Fix lint issues that now appear**.

---

### Related PRs

- https://github.com/stackabletech/listener-operator/pull/222
- https://github.com/stackabletech/secret-operator/pull/498

### CODE REVIEW

File: template/docker/Dockerfile.j2 [hunk 1]
```diff
@@ -30,9 +36,9 @@ RUN microdnf update -y --setopt=install_weak_deps=0 \
 COPY LICENSE /licenses/LICENSE
 
 COPY --from=builder /app/* /usr/local/bin/
-{[% if operator.include_productconfig is undefined or operator.include_productconfig == true %}]
-COPY deploy/config-spec/properties.yaml /etc/stackable/{[ operator.name }]/config-spec/properties.yaml
-{[% endif %}]
+# {[% if operator.include_productconfig is undefined or operator.include_productconfig == true %}]
```
Comment (lfrancke) [inline template/docker/Dockerfile.j2:39]: Are these `#` intentional?

File: template/docker/Dockerfile.j2 [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
 # =============
-FROM oci.stackable.tech/sdp/ubi9-rust-builder AS builder
+# We want to automatically use the latest. We also don't tag our images with a version.
+# hadolint ignore=DL3007
+FROM oci.stackable.tech/sdp/ubi9-rust-builder:latest AS builder
 
-FROM registry.access.redhat.com/ubi9/ubi-minimal AS operator
+# We want to automatically use the latest.
+# hadolint ignore=DL3007
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest AS operator
```
Comment (lfrancke) [inline template/docker/Dockerfile.j2:11]: Actually I'd prefer to pin this and update via renovate but that's a separate PR :)

File: template/docker/Dockerfile.j2 (hunk 1 above)
Comment (NickLarsenNZ) [reply template/docker/Dockerfile.j2:39]: Yeah, they stop the linter failing the parse the Dockerfile.

File: template/docker/Dockerfile.j2 (hunk 2 above)
Comment (NickLarsenNZ) [reply template/docker/Dockerfile.j2:11]: Perhaps we should tag the rust-builder images with a semver, unless Renovate can find the SHA (or the short commit hash which is tagged)?

Do we need to create an issue for this, or will it surface naturally?

File: template/docker/Dockerfile.j2 (hunk 1 above)
Comment (NickLarsenNZ) [reply template/docker/Dockerfile.j2:39]: I also had to remove the spaces in `{[ operator.name }]` on the following line.

Comment (lfrancke) [reply template/docker/Dockerfile.j2:39]: Thank you

File: template/docker/Dockerfile.j2 (hunk 2 above)
Comment (lfrancke) [reply template/docker/Dockerfile.j2:11]: I have an issue to "renovateify" everything....

**Review by lfrancke — APPROVED** [review]

**Pull request merged**